tests/rptest/tests/transactions_test.py

```diff
@@ -510,6 +510,8 @@ def leader_is_changed():
                        backoff_sec=2,
                        err_msg="Failed to establish current leader")
 
+        # Issue a leadership transfer
+        graceful_transfer()
```

Comment:
Can you make this test change a separate commit ?

Reply:
I prefer having tests committed together with code as they are part of context for why a change was introduced (together with the commit message). If these were 2 separate commits, then running a line blame for newly introduced `set_mem` wouldn't have this context and would leave the user with additional questions and work to do to answer questions like: "how did he knew it was broken?", "how did he prove it is fixed?". Similar argument for blaming from test file too.

What would be the argument for splitting it here? And, would the test introducing commit be before or after the tm_stm.cc change?

Reply:
usually we prefer not to have unrelated changes in one commit, but i think you are right here it is actually related and limited.